src/sicmutils/euclid.cljc

```
@@ -55,6 +57,30 @@
                   a
                   (recur b (g/remainder a b))))))
 
+(defn nearest-int [x]
+  "Rounds x to the closest integer."
+  (cond (> (g/fractional-part x) 0.5) (g/ceiling x)
+        :else (g/floor x)))
+
+(defn round-complex [z]
```

We only want to do this if we are already VERY close to a gaussian integer. Use `v/almost-integral?` on the real and imaginary parts to check.

---

What should we return for the case 
```clojure
(not (and (is-gaussian-integer a) (is-gaussian-integer b)))
```
?

---

Sorry, I like the idea of rounding as a function!  I meant that we should throw an exception in `gcd-complex` if we don't have a gaussian integer already.

---

Added throws - let me know if the message seems fine